node/hyperbahn/index.js

```diff
@@ -362,12 +362,22 @@ function advertise(opts) {
 };
 
 HyperbahnClient.prototype.unadvertise =
-function unadvertise() {
+function unadvertise(opts) {
     var self = this;
-    self.destroy();
+    self.sendRequest(opts, 'unad', unadvertiseInternalCb);
+    timers.clearTimeout(self._advertisementTimer);
+    timers.clearTimeout(self.advertisementTimeoutTimer);
     self.latestAdvertisementResult = null;
     self.state = States.UNADVERTISED;
     self.emit('unadvertised');
+    function unadvertiseInternalCb(error, result) {
+        self.logger.debug('HyperbahnClient: unadvertisement complete', {
```

Comment:
Oh, I missed this this for all cases. You may want to do logger.warn for error cases.